central/image/service/http_handler.go

```diff
@@ -0,0 +1,136 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+	clusterUtil "github.com/stackrox/rox/central/cluster/util"
+	"github.com/stackrox/rox/central/role/sachelper"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/images/enricher"
+	"github.com/stackrox/rox/pkg/images/integration"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	NumberBytes = 100
+)
+
+type sbomRequestBody struct {
+	Cluster   string `json:"cluster"`
+	ImageName string `json:"image_name"`
+	Force     bool   `json:"force"`
+}
+
+type httpHandler struct {
+	integration      integration.Set
+	enricher         enricher.ImageEnricher
+	clusterSACHelper sachelper.ClusterSacHelper
+}
+
+var _ http.Handler = (*httpHandler)(nil)
+
+// Handler returns a handler for policy http requests
+func Handler(integration integration.Set, enricher enricher.ImageEnricher, clusterSACHelper sachelper.ClusterSacHelper) http.Handler {
+	return httpHandler{
+		integration:      integration,
+		enricher:         enricher,
+		clusterSACHelper: clusterSACHelper,
+	}
+
```

Comment:
other handlers take in dependencies as argument
example /api/splunk/ta/vulnmgmt
will be easier for passing mocks inn unit tests

Reply:
> other handlers take in dependencies as argument
> example /api/splunk/ta/vulnmgmt

That is just one - there are 26 unique handlers, 10 pass a datasource, singleton, etc. 16 do not. Of the 16 that do not, 10 of them abstract the initialization away from `main.go` and setup the datasources behind the abstraction - so its split evenly on those that do and do not, such as: 

https://github.com/stackrox/stackrox/blob/e3e7847a28badb57078175b7dd4aa50afe244fbb/central/complianceoperator/v2/scanconfigurations/service/download.go#L26-L34

> will be easier for passing mocks inn unit tests

Each of the custom route handlers are [httptest](https://pkg.go.dev/net/http/httptest) compatible, so no 'mocks' needed for the routes.  From within the image service package, the `sbomHttpHandler` can be used directly with the desired mocks as needed for testing.  Whether the datasources are passed in `main.go` or not I don't believe  makes testing / mocking easier/harder (current state).

Because it's split I'll leave it to you decide, I personally find abstracting it away cleaner - but are certainly pros/cons to each approach.

Reply:
For this review I will keep it as it is